core/Pos/Binary/Core/Block.hs

```diff
@@ -26,22 +25,22 @@ instance ( Bi (T.BHeaderHash b)
          , T.BlockchainHelpers b
          ) =>
          Bi (T.GenericBlockHeader b) where
-    sizeNPut = labelS "GenericBlockHeader" $
-        putConst protocolMagic <>
-        putField T._gbhPrevBlock <>
-        putField T._gbhBodyProof <>
-        putField T._gbhConsensus <>
-        putField T._gbhExtra
-    get = label "GenericBlockHeader" $ do
-        blockMagic <- get
-        when (blockMagic /= protocolMagic) $
-            fail $ "GenericBlockHeader failed with wrong magic: " <> show blockMagic <>
-                   ", our magic is : " <> show protocolMagic
-        prevBlock <- get
-        bodyProof <- get
-        consensus <- get
-        extra <- get
-        eitherToFail $ T.recreateGenericHeader prevBlock bodyProof consensus extra
+  encode bh =  encodeListLen 5
+            <> encode protocolMagic
+            <> encode (T._gbhPrevBlock bh)
+            <> encode (T._gbhBodyProof bh)
+            <> encode (T._gbhConsensus bh)
+            <> encode (T._gbhExtra bh)
+  decode = do
+    enforceSize "GenericBlockHeader b" 5
+    blockMagic <- decode
+    when (blockMagic /= protocolMagic) $
+        fail $ "GenericBlockHeader failed with wrong magic: " <> show blockMagic
+    prevBlock <- decode
```

Comment:
I might come across as an heretic, but I think without perf measuring I cannot answer, so my inclination is usually add bang patterns where it's obvious (aka prim types and accumulators in loops) but not bother _too much_ up until I profile. 😜